scripts/run_tutorials.py

```diff
@@ -56,12 +68,18 @@ def run_script(script: str, env: Optional[Dict[str, str]] = None) -> None:
             tmp_script.write(script)
     if env is not None:
         env = {**os.environ, **env}
+    if sys.platform != "darwin":
+        # limiting memory usage fails on Mac (at least Apple Silicon)
+        preexec_fn = partial(_limit_memory, 4 * 1024 * 1024)  # 4GB memory limit
```

Comment:
Supposedly these machines should have at least 7 GB https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners

Reply:
Also I'm not sure this is in bytes? When I run

```bash
python -c "import resource; print(resource.getrlimit(resource.RLIMIT_AS))"
```

then I get `(9223372036854775807, 9223372036854775807)`.

When I run

```bash
python -c "import resource; resource.setrlimit(resource.RLIMIT_AS, (1024 * 1024 * 4, resource.RLIM_INFINITY))"
```

I get an error:

```bash
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ValueError: current limit exceeds maximum limit
```